packages/schema/src/Schema.ts

```diff
@@ -2035,13 +2082,29 @@ export const optional: {
         return optionalToRequired(
           NullishOr(schema),
           OptionFromSelf(typeSchema(schema)),
-          { decode: option_.filter<A | null | undefined, A>((a): a is A => a != null), encode: identity }
+          {
+            decode: option_.filter<A | null | undefined, A>((a): a is A => a != null),
+            encode: (t) => {
+              if ("onNoneEncoding" in options) {
+                return option_.orElse(t, () => option_.some(options.onNoneEncoding))
+              }
+              return t
+            }
+          }
         )
       } else {
         return optionalToRequired(
           UndefinedOr(schema),
           OptionFromSelf(typeSchema(schema)),
-          { decode: option_.filter(Predicate.isNotUndefined<A | undefined>), encode: identity }
+          {
+            decode: option_.filter(Predicate.isNotUndefined<A | undefined>),
+            encode: (t) => {
+              if ("onNoneEncoding" in options) {
+                return option_.orElse(t, () => option_.some(options.onNoneEncoding as undefined))
```

Comment:
This `as` is safe, because the types of option prevents setting `onNoneEncoding` to null when `nullable` is not set.
I don't know the policy for the `as`. Tell me if you think it's wrong